src/backend/allocator/gbm.rs

```diff
@@ -0,0 +1,140 @@
+//! Module for Buffers created using [libgbm](reexports::gbm)
+
+use super::{
+    dmabuf::{AsDmabuf, Dmabuf},
+    Allocator, Buffer, Format, Fourcc, Modifier,
+};
+use gbm::{BufferObject as GbmBuffer, BufferObjectFlags, Device as GbmDevice};
+use std::os::unix::io::AsRawFd;
+
+impl<A: AsRawFd + 'static, T> Allocator<GbmBuffer<T>> for GbmDevice<A> {
+    type Error = std::io::Error;
+
+    fn create_buffer(&mut self, width: u32, height: u32, format: Format) -> std::io::Result<GbmBuffer<T>> {
+        if format.modifier == Modifier::Invalid || format.modifier == Modifier::Linear {
+            let mut usage = BufferObjectFlags::SCANOUT | BufferObjectFlags::RENDERING;
+            if format.modifier == Modifier::Linear {
+                usage |= BufferObjectFlags::LINEAR;
+            }
+            self.create_buffer_object(width, height, format.code, usage)
+        } else {
+            self.create_buffer_object_with_modifiers(
+                width,
+                height,
+                format.code,
+                Some(format.modifier).into_iter(),
+            )
+        }
+    }
+}
+
+impl<T> Buffer for GbmBuffer<T> {
+    fn width(&self) -> u32 {
+        self.width().unwrap_or(0)
+    }
+
+    fn height(&self) -> u32 {
+        self.height().unwrap_or(0)
+    }
+
+    fn format(&self) -> Format {
+        Format {
+            code: self.format().unwrap_or(Fourcc::Argb8888), // we got to return something, but this should never happen anyway
+            modifier: self.modifier().unwrap_or(Modifier::Invalid),
+        }
+    }
+}
+
+/// Errors during conversion to a dmabuf handle from a gbm buffer object
+#[derive(thiserror::Error, Debug)]
+pub enum GbmConvertError {
+    /// The gbm device was destroyed
+    #[error("The gbm device was destroyed")]
+    DeviceDestroyed(#[from] gbm::DeviceDestroyedError),
+    /// The buffer consists out of multiple file descriptions, which is currently unsupported
+    #[error("Buffer consists out of multiple file descriptors, which is currently unsupported")]
+    UnsupportedBuffer,
+    /// The conversion returned an invalid file descriptor
+    #[error("Buffer returned invalid file descriptor")]
+    InvalidFD,
+}
+
+impl<T> AsDmabuf for GbmBuffer<T> {
+    type Error = GbmConvertError;
+
+    fn export(&self) -> Result<Dmabuf, GbmConvertError> {
+        let planes = self.plane_count()? as i32;
+
+        //TODO switch to gbm_bo_get_plane_fd when it lands
```

Comment:
Also, once the PR is merged, we should probably open an issue to track this TODO and not forget it.